### ISSUE TITLE
Objects being mutated within template function

### ISSUE THREAD

**Issue by edwilliams** [webpro/dyson#105]
**data.js**
```javascript
module.exports = ['foo', 'bar']
```

**index.js**
```javascript
const data = require('./data')

console.log(data); // ['foo', 'bar']

module.exports = {
  path: '/path/to/api',
  template: () => {
    console.log(data); // ['foo', 'bar', method: undefined ]
    return {}
  }
};
```

...this is also happening within `status` the function too:

```javascript
status: (req, res, next) => {
  console.log(data); // ['foo', 'bar', method: undefined ]
  next()
}
```

**Comment by edwilliams** [comment]
_...temporary workaround using `clone`_

```javascript
const data = require('./data');

const clone = val => JSON.parse(JSON.stringify(val)); // ...or any clone func

console.log(data); // ['foo', 'bar']

module.exports = {
  path: '/path/to/api',
  template: () => {
    console.log(data); // ['foo', 'bar', method: undefined ]
    console.log(clone(data)); // ['foo', 'bar' ]
    return {};
  }
};

```

**Comment by webpro** [comment]
dyson treats (the exported object of) all files in the folder as configuration objects, and always sets a `method`. So I would advise to put any modules that are not js files outside this directory.

However, dyson should not mutate the configuration, so I've just released [3.0.2](https://github.com/webpro/dyson/releases/tag/3.0.2) that fixes this issue.

**Comment by edwilliams** [comment]
Cheers @webpro, that's fixed it!
